glom/core.py

```diff
@@ -753,6 +753,90 @@ def __repr__(self):
         return '%s(%r, args=%r, kwargs=%r)' % (cn, self.func, self.args, self.kwargs)
 
 
+class Partial(object):
+    """
+    makes "upgrading" arbitrary python functions into
+    glom specs more convenient, so that itertools, toolz, et al
+    may be more easily integrated into streaming glom chains
+    """
+    def __init__(self, func, specfunc=None):
+        self.func, self.subspec = func, specfunc
```

Comment:
Should we put some type verification in here? We could require func be wrapped in `Spec()` if that's a more explicit way of indicating a spec func (there's also the classmethod to make that convenient)

Reply:
the intention is that the public API for adding a specfunc is the classmethod ... it's actually a little bit unfortunate that specfunc is part of the signature here, maybe there's a way to swizzle it around so that isn't the case?